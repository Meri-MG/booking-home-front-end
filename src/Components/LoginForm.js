import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router';
import { logUserIn } from '../Redux/Login/Login';

const LoginForm = () => {
  const dispatch = useDispatch();
  const navigateTo = useNavigate();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const { status, IsLogged_in } = useSelector((state) => state.LogIn);
  if (status === 'created' || IsLogged_in)navigateTo('/');

  const handleOnSubmit = (event) => {
    event.preventDefault();
    const user = { email, password };
    dispatch(logUserIn({ user }));
  };

  return (
    <div className="flex w-1/2 justify-center items-center bg-white">
      <form onSubmit={handleOnSubmit} className="bg-white">
        <h1 className="text-gray-800 font-bold text-2xl mb-1">Hello Again!</h1>
        <p className="text-sm font-normal text-gray-600 mb-7">Welcome Back</p>
        {status === 401 && (<p className="text-lg text-red-500 font-bold break-words p-4 max-w-fit">Invalid email or Password please check your information and try again</p>)}
        <div className="flex items-center border-2 py-2 px-3 rounded-2xl mb-4">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-5 w-5 text-gray-400"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              d="M16 12a4 4 0 10-8 0 4 4 0 008 0zm0 0v1.5a2.5 2.5 0 005 0V12a9 9 0 10-9 9m4.5-1.206a8.959 8.959 0 01-4.5 1.207"
            />
          </svg>
          <input
            className="pl-2 outline-none border-none"
            type="email"
            name="email"
            value={email}
            id="email"
            placeholder="Email Address"
            onChange={(event) => setEmail(event.target.value)}
            maxLength="70"
            required
          />
        </div>
        <div className="flex items-center border-2 py-2 px-3 rounded-2xl">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-5 w-5 text-gray-400"
            viewBox="0 0 20 20"
            fill="currentColor"
          >
            <path
              fillRule="evenodd"
              d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
              clipRule="evenodd"
            />
          </svg>
          <input
            className="pl-2 outline-none border-none"
            type="password"
            name="Password"
            value={password}
            id="password"
            onChange={(event) => setPassword(event.target.value)}
            placeholder="Password"
            maxLength="70"
            required
          />
        </div>
        <button
          type="submit"
          className="block w-full bg-main-color mt-4 py-2 rounded-2xl text-white font-semibold mb-2"
        >
          Login
        </button>
      </form>
    </div>
  );
};

export default LoginForm;
