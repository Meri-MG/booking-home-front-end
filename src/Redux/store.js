import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import loginReducer from './Login/Login';
import registerReducer from './Register/Register';
import apartmentsReducer from './Apartments/Apartments';
import apartmentsDetailsReducer from './Details/Details';
import reservationsReducer from './Reserved/Reserved';

const reducers = combineReducers({
  LogIn: loginReducer,
  Register: registerReducer,
  Apartments: apartmentsReducer,
  Details: apartmentsDetailsReducer,
  Reservations: reservationsReducer,
});

const middlewares = [thunk, logger];

const store = createStore(reducers, applyMiddleware(...middlewares));

export default store;
