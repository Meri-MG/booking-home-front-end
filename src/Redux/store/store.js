import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const reducers = combineReducers({
});

const middlewares = [thunk,logger];

const store = createStore(reducers, applyMiddleware(...middlewares));

export default store;
