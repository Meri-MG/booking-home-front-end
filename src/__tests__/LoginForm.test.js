import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/store';
import LoginForm from '../Components/LoginForm';

it('renders login form page correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <LoginForm />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
