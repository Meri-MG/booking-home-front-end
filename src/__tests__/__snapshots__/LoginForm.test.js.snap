// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login form page correctly 1`] = `
<div
  className="flex w-1/2 justify-center items-center bg-white"
>
  <form
    className="bg-white"
    onSubmit={[Function]}
  >
    <h1
      className="text-gray-800 font-bold text-2xl mb-1"
    >
      Hello Again!
    </h1>
    <p
      className="text-sm font-normal text-gray-600 mb-7"
    >
      Welcome Back
    </p>
    <div
      className="flex items-center border-2 py-2 px-3 rounded-2xl mb-4"
    >
      <svg
        className="h-5 w-5 text-gray-400"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 12a4 4 0 10-8 0 4 4 0 008 0zm0 0v1.5a2.5 2.5 0 005 0V12a9 9 0 10-9 9m4.5-1.206a8.959 8.959 0 01-4.5 1.207"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
      <input
        className="pl-2 outline-none border-none"
        id="email"
        maxLength="70"
        name="email"
        onChange={[Function]}
        placeholder="Email Address"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="flex items-center border-2 py-2 px-3 rounded-2xl"
    >
      <svg
        className="h-5 w-5 text-gray-400"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
          fillRule="evenodd"
        />
      </svg>
      <input
        className="pl-2 outline-none border-none"
        id="password"
        maxLength="70"
        name="Password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <button
      className="block w-full bg-main-color mt-4 py-2 rounded-2xl text-white font-semibold mb-2"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
